### PR TITLE
Add last_value window function

### PR DESCRIPTION
**Pull request by pramodsatya**

### CODE REVIEW

File: velox/functions/prestosql/window/LastValue.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  // Argument index of the first_value argument column in the input row vector.
+  // This is used to retrieve column values from the partition data.
+  column_index_t valueIndex_;
+
+  const exec::WindowPartition* partition_;
+
+  // This offset tracks how far along the partition rows have been output.
+  // This can be used to optimize reading offset column values corresponding
+  // to the present row set in getOutput.
+  vector_size_t partitionOffset_;
```
Comment (aditi-pandit) [inline velox/functions/prestosql/window/LastValue.cpp:68]: Is this variable needed ?

**Review by aditi-pandit — COMMENTED** [review]
This too is looking alright. Let's do some frame related PRs before further work on this.

File: velox/functions/prestosql/window/LastValue.cpp (hunk 1 above)
Comment (pramodsatya) [reply velox/functions/prestosql/window/LastValue.cpp:68]: Thanks for pointing it out, removed it.

**Comment by aditi-pandit** [comment]
> This too is looking alright. Let's do some frame related PRs before further work on this.

I feel that you could try to combine first_value and last_value in the same class since the logic is very similar. Use a boolean template variable to trigger the respective behavior. You can work across 2 PRs though.

**Comment by pramodsatya** [comment]
Thanks for the suggestion, combined these changes with #2933 and closing this PR.

**Pull request closed without merging**